src/Cake.Common/Build/TeamCity/TeamCityDisposableHelpers.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Cake.Common.Build.TeamCity
+{
+    /// <summary>
+    /// Disposable helper for writing TeamCity message blocks.
+    /// </summary>
+    public sealed class TeamCityBlock : IDisposable
```

Comment:
In regards to 1 & 2: Thinking about it now, I think I may just do what I've seen down elsewhere, and make a `DisposableHelper` of some sort, that wraps an action in an a Disposable wrapper. Could be used in other places then. That would move startblock out and remove the need for both these wrappers.

Regarding 3: I purposely kept `WriteEndBlock` available on the TC provider for that reason. If they're doing something that deviates from the normal `using`, they should just use the `Start` and `End` methods.